rcon/automods/seeding_rules.py

```diff
@@ -257,16 +259,40 @@ def punitions_to_apply(
                 if server_player_count >= dwc.max_players:
                     self._disable_for_round("disallowed_weapons")
 
+                ecf = self.config.enforce_cap_fight
+                if server_player_count >= ecf.max_players:
+                    self._disable_for_round("enforce_cap_fight")
+
                 violations = []
                 if (
-                    not self._is_seeding_rule_disabled("disallowed_roles")
-                    and drc.min_players <= server_player_count < drc.max_players
+                        not self._is_seeding_rule_disabled("disallowed_roles")
+                        and drc.min_players <= server_player_count < drc.max_players
                 ):
                     if aplayer.role in drc.roles:
                         violations.append(
                             drc.message.format(role=drc.roles.get(aplayer.role))
                         )
 
+                if not self._is_seeding_rule_disabled("enforce_cap_fight") and \
+                        (team == "axis" and game_state["axis_score"]) >= ecf.max_caps or \
+                        (team == "allies" and game_state["allied_score"] >= ecf.max_caps):
+                    self.logger.debug("Player is on " + team + " side and winning")
+                    op = player['offense']
+                    oop = watch_status.offensive_points.setdefault(aplayer.name, 0)
+
+                    self.logger.debug(
+                        "Player had " + str(oop) + " offensive points last and now " + str(op))
+                    if oop != 0 and oop < op:
+                        violations.append(ecf.message)
+
+                        if ecf.skip_warning:
+                            warnings = watch_status.warned.setdefault(aplayer.name, [])
+                            for _ in range(self.config.number_of_warning):
+                                warnings.append(datetime.now() - timedelta(seconds=self.config.warning_interval_seconds + 1))
+                    watch_status.offensive_points[aplayer.name] = op
```

Comment:
Why are we using player names and not steam IDs as key here?

Reply:
good question, I actually don't know, it was that case when I started working with the existing automod.
However, I think it might be worth revisiting this and use steam IDs instead. But, I would not do that in this PR and instead in a separate one, what do you think?

Reply:
Sure thing